src/Core/OrganizationFeatures/AuthorizationHandlers/GroupAuthorizationHandler.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.OrganizationFeatures.AuthorizationHandlers;
+
+class GroupAuthorizationHandler : AuthorizationHandler<GroupOperationRequirement, Group>
+{
+    private readonly ICurrentContext _currentContext;
+
+    public GroupAuthorizationHandler(ICurrentContext currentContext)
+    {
+        _currentContext = currentContext;
+    }
+
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        GroupOperationRequirement requirement,
+        Group resource)
+    {
+        // Currently all GroupOperationRequirements have the same permission requirements
+        if (requirement == GroupOperations.Create ||
+            requirement == GroupOperations.Read ||
+            requirement == GroupOperations.Update ||
+            requirement == GroupOperations.Delete ||
+            requirement == GroupOperations.AddUser ||
+            requirement == GroupOperations.DeleteUser)
+        {
+            CanManageGroups(context, requirement, resource);
+        }
```

Comment:
I think this is a good example of the logic being backwards.

What this does is take a requirement and map that requirement to a permission. The questions we're more likely to ask are "who can read a group", not necessarily "what does manage groups get me?"

To that end, I would like this to be a switch on `GroupOperations` that shims out to one method per requirement. Each requirement can then evolve as the permission model for that requirement changes

Reply:
I've changed this to use a switch statement with 1 private method per requirement, even if they ultimately redirect to the same logic (`CanManage`). I think this is what you were after, per your code examples above.